### PR TITLE
[11.0][IMP] Remove arbitrary file upload limit (25Mb) and turn it to an option

### PR DESCRIPTION
**Pull request by zakiuu**
Back port of 402379a48a31bc486145e81d8dae2910ae446b48

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/fields/basic_fields.js [hunk 1]
```diff
@@ -1147,7 +1147,12 @@ var AbstractFieldBinary = AbstractField.extend({
         this._super.apply(this, arguments);
         this.fields = record.fields;
         this.useFileAPI = !!window.FileReader;
-        this.max_upload_size = 25 * 1024 * 1024; // 25Mo
+        // Use the fields options to determine the max upload size
+        try {
+            this.max_file_size = utils.parse_human_size(this.attrs.options.max_file_size);
```
Comment (fmdl) [inline addons/web/static/src/js/fields/basic_fields.js]: `this.max_upload_size = utils.parse_human_size(this.attrs.options.max_file_size);`

**Comment by mart-e** [comment]
Hi @zakiuu,

Thanks for the proposal.
If you are backporting a commit, please try to keep the authorship. You can make the cherry-pick and mention the source in the commit message.
Also, it seems you have not signed the CLA. Could you add it if we want to integrate your patch?

@ged-odoo it seems that 402379a48a31bc486145e81d8dae2910ae446b48 was not kept during the move at 905e01921f3c3ef43ace6fc537d1d8c0d280002
Any reason to remove this option?

**Comment by ged-odoo** [comment]
@mart-e No reason at all.  My guess is that it was lost during our 'rebase of hell'. :/

**Comment by zakiuu** [comment]
@mart-e thanks for the review and sorry for the late response, I'm afraid that cherry-picking 402379a doesn't work as the file structure in v11 and v12 isn't the same.
```
error: could not apply 402379a48a3... [IMP] Remove arbitrary file upload limit (25Mb) and turn it to an option (Fixes #14824)
hint: after resolving the conflicts, mark the corrected paths
hint: with 'git add <paths>' or 'git rm <paths>'
hint: and commit the result with 'git commit'
```
For the my CLA it's already there so I don't why it's not showen: https://github.com/odoo/odoo/blob/11.0/doc/cla/individual/zakiuu.md

**Comment by mart-e** [comment]
@zakiuu for the CLA, it is because you have signed it with a gmail address but this PR is signed with `@tajawal.com`.

**Comment by vedler** [comment]
Should we make a new PR if this is still not merged/rebased properly? I think this whole issue (#14824) has been in limbo for almost 4 years now.

**Comment by C3POdoo** [comment]
Dear @zakiuu,

Thank you for your contribution but the version 11.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**